src/SigasiProjectCreator/ProjectOptions.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+"""
+    :copyright: (c) 2008-2023 Sigasi
+    :license: BSD, see LICENSE for more details.
+"""
+import argparse
+import os
+import pathlib
+
+from SigasiProjectCreator import VerilogVersion, VhdlVersion
+
+
+class ProjectOptions:
+    def __init__(self, args_list=None):
+        parser = argparse.ArgumentParser(prog='SigasiProjectCreator')
+        parser.add_argument('project_name', help='Project name')
+        parser.add_argument('input_file', help='Input file or comma-separated list of input files', nargs='+')
+        parser.add_argument('-d', '--destination', action='store', dest='destination_folder',
+                            help='Root folder of created project', type=pathlib.Path, default=pathlib.Path.cwd())
+        parser.add_argument('-l', '--layout', action='store', dest='layout',
+                            choices=['in-place', 'simulator', 'linked-files-flat', 'linked-files-tree',
+                                     'linked-folders'], default='in-place',
+                            help='Project layout: in-place (default), simulator (one folder per library with '
+                                 'linked files), linked-files-flat (one folder with links to all files), '
+                                 'linked-files-tree (virtual folders like the source tree, with links to files), '
+                                 'or linked-folders (mix of virtual and linked folders)')
```

Comment:
To avoid the duplication between here and the dictionary in `get_project_creator`, I'd automatically generate this list based on registered project creators. Something likes this:

```python
from argparse import ArgumentParser
import sys

project_creators = {}
def project_creator(key):
	def register(cls):
		project_creators[key] = cls
		return cls
	return register

@project_creator('in-place')
class ProjectCreatorInPlace():
	"""default"""
	pass

@project_creator('linked-files-flat')
class LinkedFilesFlat():
	"""one folder with links to all files"""
	pass

parser = ArgumentParser(prog="test")
parser.add_argument('-l', '--layout',
    choices=project_creators.keys(),
    help=('Any of the following layouts: ' + ', '.join(f'{key} ({cls.__doc__})' for key, cls in project_creators.items())))
parser.parse_args(sys.argv[1:])
```

Reply:
You could also only show the first line of the docstring in the help and include more documentation as to what the layout plans to do. A `--help-layouts` or similar could then print that complete docstring.

Reply:
Implemented